### ISSUE TITLE
Accessibility: Missing heading

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#51892]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The cart page does not have any headings for the main sections of the page, which may make it more difficult to navigate for screen reader users. The only heading output on the page is an H2 for the "You may be interested in" section (if it is present).

![Image](https://github.com/user-attachments/assets/b4ed4357-ff21-4bb6-b410-6ddbc158d22d)

### Component or Page: 

Page: Cart (Modern)  

### Name affected element:

Products in Cart Section

### Relevant code snippets:

N/A

### Recommended fix:

Add a screen reader only H2 above the cart items table that reads "Products in Cart" and correct the "cart totals" heading to be an H2 as reported in #WooCore-028. This will result in a heading structure of the page that is more meaningful and easier to navigate with a screen reader.
